docs/spec/spec.md

```diff
@@ -0,0 +1,416 @@
+# ```
+# check idocListener.detach(service);
+# ```
+#
+# + `s` - The service to be detached.
+# + `return` - An error if the detachment fails or else `()`.
+public isolated function detach(Service s) returns error? = @java:Method {
+    'class: "io.ballerina.lib.sap.Listener"
+} external;
+```
+
+- To start the iDoc listener, the `'start` function can be used.
+
+```ballerina
+# Starts the `iDocListener`.
+# ```
+# check idocListener.'start();
+# ```
+#
+# + `return` - An error if the start fails or else `()`.
+public isolated function 'start() returns error? = @java:Method {
+    'class: "io.ballerina.lib.sap.Listener"
+} external;
+```
+
+- To stop the iDoc listener gracefully, the `gracefulStop` function can be used.
+
+```ballerina
+# Stops the `iDocListener` gracefully.
+# ```
+# check idocListener.gracefulStop();
+# ```
+#
+# + `return` - An error if the graceful stop fails or else `()`.
+public isolated function gracefulStop() returns error? = @java:Method {
+    'class: "io.ballerina.lib.sap.Listener"
+} external;
+```
+
+- To stop the iDoc listener immediately, the `immediateStop` function can be used.
+
+```ballerina
+# Stops the `iDocListener` immediately.
+# ```
+# check idocListener.immediateStop();
+# ```
+#
+# + `return` - An error if the immediate stop fails or else `()`.
+public isolated function immediateStop() returns error? = @java:Method {
+    'class: "io.ballerina.lib.sap.Listener"
+} external;
+```
+
+- To initialize the listener with a given configuration, the `init` function can be used.
+
+```ballerina
+# Initializes a `Listener` object with the given configuration, which is used to listen to iDoc messages.
+# ```
+# check idocListener.init(config);
+# ```
+#
+# + `configurations` - The required configuration for initializing the listener.
+# + `serverName` - Name of the server (optional, default is a UUID).
+# + `return` - An error if initialization fails or else `()`.
+public isolated function init(ServerConfig|AdvancedConfig serverConfig, string serverName = uuid:createType4AsString()) returns Error? {
+    return externInit(self, serverConfig, serverName);
+}
+```
+
+#### 2.2.3 Available Operations
+
+##### 2.2.3.1 Receive IDocs
+
+The Listener can be used to receive IDocs from the SAP system, which can then be processed within your Ballerina application.
+
+```ballerina
+listener jco:Listener iDocListener = new (configurations);
+
+service jco:Service on iDocListener {
+    remote function onReceive(xml iDoc) returns error? {
+        check io:fileWriteXml("resources/received_idoc.xml", iDoc);
+        io:println("IDoc received and saved.");
+    }
+}
+```
+
+When an IDoc is received, it will be in XML format, and the user can easily map it to a record using Ballerina's XML-to-Record conversion features.
+
+##### 2.2.3.2 Error Handling
+
+ Listener
+
+The Listener also supports error handling, allowing you to capture and manage any issues that occur during IDoc reception.
```

Comment:
Is this word added intentionally? looks redundant to me
```suggestion
The Listener also supports error handling, allowing you to capture and manage any issues that occur during IDoc reception.
```

Reply:
Changed with [0553b90](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/56/commits/0553b901a67bf7a546060ea221052c9ae955cb00).